### PR TITLE
move dockerfile to server repo

### PR DESCRIPTION
**Pull request by cpanato**
As we discussed in the docker channel in community.mattermost.com
We decide to move the docker file to the server repo and apply some simplification to the image as well

this PR adds the dockerfile to the server repo

- [x] Add dockerFile 
- [x] tested locally with docker compose
- [x] Test the image generated in a test server
- [x] update the build pipeline to use this dockerfile to build the image for PRs and branches
- [ ] internal: update the release pipeline to use this file when releasing mattermost - in a followup after this PR gets merged

---
I would like some feedback

### CODE REVIEW

**Comment by cpanato** [comment]
cc @pichouk

**Comment by cpanato** [comment]
@jasonblais this the initial effort to move the docker file to here and deprecate the docker repo and the prod-app image as well

File: build/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+if [ "${1:0:1}" = '-' ]; then
+    set -- mattermost "$@"
+fi
+
+exec "$@"
```
Comment (pichouk) [inline build/entrypoint.sh]: No newline at the end of the file, I guess it's not on purpose.

File: build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  CMD curl -f http://localhost:8065/api/v4/system/ping || exit 1
+
+
+# Configure entrypoint and command
+COPY entrypoint.sh /
+ENTRYPOINT ["/entrypoint.sh"]
+WORKDIR /mattermost
+CMD ["mattermost"]
+
+EXPOSE 8065 8067 8074 8075
```
Comment (pichouk) [inline build/Dockerfile]: Why all those ports ?

**Comment by pichouk** [comment]
LGTM about this, it's a common practice to have Dockerfile in the same repository as the code base.

File: build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
+FROM alpine:3.10
+
+# Some ENV variables
+ENV PATH="/mattermost/bin:${PATH}"
+ARG PUID=2000
+ARG PGID=2000
+ARG MM_PACKAGE=
```
Comment (crspeller) [inline build/Dockerfile]: If we have this in the main repo are people going to expect a docker build and run to just work? It won't because of this.

Comment (cpanato) [reply build/Dockerfile]: fixed

File: build/Dockerfile (hunk 2 above)
Comment (cpanato) [reply build/Dockerfile]: they are for the gossip/metrics/cluster

File: build/entrypoint.sh (hunk 1 above)
Comment (cpanato) [reply build/entrypoint.sh]: done

**Comment by cpanato** [comment]
docker image : `mattermost/mattermost-enterprise-edition:0dd3a46`

**Comment by cpanato** [comment]
image is running smoothly in the test server

**Comment by cpanato** [comment]
Ready for final reviews

File: build/Jenkinsfile.branch [hunk 4]
```diff
@@ -117,10 +117,16 @@ pipeline {
 
     stage ('Build Docker mattermost-enterprise-edition') {
       steps {
-        script {
-          platformStages.platformCheckoutDockerRepo('master')
-          def MMBINARYPATH="https://releases.mattermost.com/mattermost-platform/" + env.BRANCH_NAME + "/mattermost-enterprise-linux-amd64.tar.gz"
-          platformStages.platformBuildDockerImage(MMBINARYPATH, env.BRANCH_NAME + '.' + env.BUILD_NUMBER + '.' + SERVER_GIT_COMMIT)
+        dir('src/github.com/mattermost/mattermost-server') {
+          withCredentials([usernamePassword(credentialsId: 'materbuild-docker-hub', usernameVariable: 'DOCKER_USER', passwordVariable: 'DOCKER_PASS')]) {
```
Comment (jwilander) [inline build/Jenkinsfile.branch]: Is `materbuild-docker-hub` a typo?

Comment (cpanato) [reply build/Jenkinsfile.branch]: it is but the typo is in jenkins. will fix and update

Comment (cpanato) [reply build/Jenkinsfile.branch]: fixed

**Pull request merged**